node/src/actors/json_rpc/server.rs

```diff
@@ -187,15 +187,53 @@ impl Handler<NewBlock> for JsonRpcServer {
                         .into_actor(self)
                         .then(move |res, _act, _ctx| {
                             if let Err(e) = res {
-                                log::error!("Failed to send notification: {:?}", e);
+                                log::error!("Failed to send block notification: {:?}", e);
                             }
 
                             actix::fut::ok(())
                         }),
                 );
             }
         } else {
-            log::error!("Failed to acquire lock in NewBlock handle");
+            log::error!("Failed to acquire lock in BlockNotify handle");
+        }
+    }
+}
+
+impl Handler<SuperBlockNotify> for JsonRpcServer {
+    type Result = ();
+
+    fn handle(&mut self, msg: SuperBlockNotify, ctx: &mut Self::Context) -> Self::Result {
+        log::debug!("Got SuperBlockNotify message, sending notifications...");
+        log::trace!(
+            "Notifying consolidation of 1 superblock and {} blocks: {:?}",
+            msg.consolidated_block_hashes.len(),
+            msg.consolidated_block_hashes
+        );
+
+        let hashes = serde_json::to_value(msg)
+            .expect("JSON serialization of SuperBlockNotify should never fail");
+        if let Ok(subscriptions) = self.subscriptions.lock() {
+            if let Some(superblocks_subscriptions) = subscriptions.get("superblocks") {
+                for (subscription, (sink, _params)) in superblocks_subscriptions {
+                    log::debug!("Sending superblock notification through sink {:?}", sink);
+                    let params = jsonrpc_core::Params::from(SubscriptionResult {
+                        result: hashes.clone(),
+                        subscription: subscription.clone(),
+                    });
+                    ctx.spawn(sink.notify(params).into_actor(self).then(move |res, _, _| {
+                        if let Err(e) = res {
+                            log::error!("Failed to send notification: {:?}", e);
+                        }
+
+                        actix::fut::ok(())
+                    }));
+                }
+            } else {
+                log::warn!("Failed to find a subscription for superblocks notifications");
```

Comment:
This warning will show when there are no clients connected. Remove it or move it to debug:
```suggestion
                log::debug!("No subscriptions for superblocks notifications");
```